-1

Tested at: a858c23a85cc4d48d701056b799d05d5482e8c4a


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/17b8235c793832707138ee5eeb04a427de5fda02
https://github.com/cms-sw/cmssw/commit/c3eacf438236abfc0812bfe44b37ee1e77b78f76
https://github.com/cms-sw/cmssw/commit/535718c9b91c490c4de2b1c73507b859b20eceb9
https://github.com/cms-sw/cmssw/commit/a91c6fa5cb01c98db4cd527935e35dcc869a279d
https://github.com/cms-sw/cmssw/commit/4899017f2c48860977d63db3bda3f147c6355c1e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25877/33024/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25877/33024/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25877/33024/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 48 COMPILER='llvm compile'

```


<pre>>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-02-06-1100/src/FWCore/Integration/test/WhatsIt.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-02-06-1100/src/FWCore/Integration/test/GadgetRcd.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-02-06-1100/src/FWCore/Integration/test/Doodad.cc 
In file included from /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-02-06-1100/src/FWCore/Integration/test/WhatsItESProducer.cc:26:
In file included from /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-02-06-1100/src/FWCore/Framework/interface/ESProducer.h:76:
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-02-06-1100/src/FWCore/Framework/interface/ESConsumesCollector.h:79:14: error: no matching member function for call to 'consumes'
      return consumes<Product, RECORD>(tag);
             ^~~~~~~~~~~~~~~~~~~~~~~~~
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-02-06-1100/src/FWCore/Integration/test/WhatsItESProducer.cc:99:22: note: in instantiation of function template specialization 'edm::ESConsumesCollectorT<GadgetRcd>::consumes<edmtest::Doodad>' requested here
  token_ = collector.consumes<edmtest::Doodad>(edm::ESInputTag{"", data_label});
                     ^
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/17b8235c793832707138ee5eeb04a427de5fda02
https://github.com/cms-sw/cmssw/commit/c3eacf438236abfc0812bfe44b37ee1e77b78f76
https://github.com/cms-sw/cmssw/commit/535718c9b91c490c4de2b1c73507b859b20eceb9
https://github.com/cms-sw/cmssw/commit/a91c6fa5cb01c98db4cd527935e35dcc869a279d
https://github.com/cms-sw/cmssw/commit/4899017f2c48860977d63db3bda3f147c6355c1e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25877/33024/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25877/33024/git-merge-result

--- a/FWCore/Framework/interface/ESConsumesCollector.h
+++ b/FWCore/Framework/interface/ESConsumesCollector.h
@@ -46,7 +46,7 @@ namespace edm {
 
     // ---------- member functions ---------------------------
     template <typename Product, typename Record>
-    auto consumes(ESInputTag const& tag) {
+    auto consumesFrom(ESInputTag const& tag) {
       return ESGetToken<Product,Record>{tag};
     }
 
@@ -72,11 +72,10 @@ namespace edm {
     ESConsumesCollectorT<RECORD>& operator=(ESConsumesCollectorT<RECORD>&&) = default;
     
     // ---------- member functions ---------------------------
-    using ESConsumesCollector::consumes;
     
     template <typename Product>
     auto consumes(ESInputTag const& tag) {
-      return consumes<Product, RECORD>(tag);
+      return consumesFrom<Product, RECORD>(tag);
     }
     
   private:

